From 4ccae651cbc718c520c26d69b951e68a2ad76f0a Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 04 Aug 2018 21:18:30 +0300
Subject: [PATCH] board: dali is a new ARM64 SoC Fixup

Update script to new flash Image
No need to have both sides we can just burn twice to different offsets
---

diff --git a/board/habanalabs/dali/post-image.sh b/board/habanalabs/dali/post-image.sh
index e42cfa9..2788511 100755
--- a/board/habanalabs/dali/post-image.sh
+++ b/board/habanalabs/dali/post-image.sh
@@ -57,23 +57,27 @@
 	make -C btl clean all ${BTL_CMD_ARGS}
 	cp btl/sectors-be.bin .
 
+	# Metadata
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-pkey-hash.bin
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-ver.bin
+	echo "11110000" > dali-next-image.txt
+	dd status=none conv=sync bs=64K if=dali-next-image.txt					 > dali-next-image.bin
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-images-pointers.bin
+
 	# BTL
 	dd status=none conv=sync bs=256K	if=sectors-be.bin				 > dali-spi-flash.bin.nocrc
-	# Primary SPL
+	# Metadata
+	dd status=none conv=sync bs=64K		if=dali-pkey-hash.bin				>> dali-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=dali-ver.bin					>> dali-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=dali-next-image.bin				>> dali-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=dali-images-pointers.bin			>> dali-spi-flash.bin.nocrc
+	# SPL
 	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
-	# Secondary SPL
-	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
-	# Primary U-Boot
-	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}				>> dali-spi-flash.bin.nocrc
-	# Secondary U-Boot
+	# U-Boot
 	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}				>> dali-spi-flash.bin.nocrc
 	# U-Boot ENV
 	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
-	# U-Boot ENV Redundand
-	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
 	# PCIe FW
-	dd status=none conv=sync bs=64K		if=${PCIE_PHY_FW_BINFILE}			>> dali-spi-flash.bin.nocrc
-	# PCIe FW Redundand
 	dd status=none conv=sync bs=64K		if=${PCIE_PHY_FW_BINFILE}			>> dali-spi-flash.bin.nocrc
 
 	# Prepare checksum
